templates/contribute/tags_and_branches.md

```diff
@@ -9,43 +9,51 @@ Mathlib.
 
 ## Things you need to know
 
-* If you are making a PR to `leanprover/lean4` which may involve breaking changes,
+* If you are making a pull request to `leanprover/lean4` which may involve breaking changes,
   please rebase your PR onto the `nightly` branch. This will enable combined CI with Mathlib.
 
-* If you are making a PR to `leanprover-community/mathlib4`,
+* If you are making a pull request to `leanprover-community/mathlib4`,
   please ask for "write" access to the repository via the [zulip chat](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/github.20permission), and push your branch to that repo.
-  This will enable Mathlib's `.olean` cache to include your PR.
+  This will enable Mathlib's `.olean` cache to include your pull request.
 
 ## Tags and branches
 
 ### `leanprover/lean4`
 
 * Development occurs on the `master` branch.
 * Stable releases and release candidates have tags, e.g. `v4.2.0` or `v4.3.0-rc1`.
-  * To use these, your `lean-toolchain` should contain e.g. `leanprover/lean4:v4.2.0`.
+  * To use these release in a project, your `lean-toolchain` should contain e.g. `leanprover/lean4:v4.2.0`.
 * Stable releases arrive at the end of each month, and are identical to the last release candidate.
 * The first release candidate of the next version is released immediately after the stable release.
 * Each version has a `releases/v4.X.0` feature branch, which may contain
   * additional commits for release notes
   * cherry picked commits from `master` for critical fixes released via release candidates.
 * We make a regular nightly release from `master`, which has a tag e.g. `nightly-2023-11-01` on the
   `leanprover/lean4-nightly` repository.
-  * To use these, your `lean-toolchain` should contain e.g. `leanprover/lean4:nightly-2023-11-01`.
+  * To use a nightly release in a project, your `lean-toolchain` should contain e.g. `leanprover/lean4:nightly-2023-11-01`.
 * There is a `nightly` branch on `leanprover/lean4` which follows the most recent commit which was
   used to construct a nightly release.
-* Every PR automatically receives a toolchain. For PR #NNNN, your `lean-toolchain` should contain
+* Every PR automatically receives a toolchain after CI completes successfully.
+  To use PR #NNNN in a project, your `lean-toolchain` should contain
   `leanprover/lean4-pr-releases:pr-release-NNNN`.
+* Every PR automatically generates a `lean-pr-testing-NNNN` branch at Mathlib,
+  described in detail below, and results from this branch are reported via comments
+  in the PR discussion.
+* For any PR that potentially affects Std or Mathlib, you should base your PR off the HEAD of the `nightly` branch.
+  The combined CI for Std and Mathlib will not run automatically unless this is the case.
+* You can determine the most recent nightly release in the history of your PR by running
+  `script/most-recent-nightly-tag.sh` in the Lean repository.
```

Comment:
Running this on my machine, I get:
```
mhuisi@DESKTOP-SES5827 ~/lean4 (mhuisi/import-auto-completion)> ./script/most-recent-nightly-tag.sh
fatal: 'nightly' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
2023-11-20
```
2023-11-20 obviously also does not exist, and I believe my branch is based off of the 2023-11-21 nightly.

Reply:
Ooof! Fixed in https://github.com/leanprover/lean4/pull/2947